### PR TITLE
Bug 1843168 - Expose Mockito through test support component

### PR DESCRIPTION
**Pull request by MatthewTighe**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1843168

### CODE REVIEW

File: android-components/components/support/test/build.gradle
Comment (rahulsainani) [inline android-components/components/support/test/build.gradle:40]: [api](https://developer.android.com/topic/modularization/patterns) leads to larger build times as multiple modules might have to be built, it's for testing but still something to consider. I see you want to expose verify functions, maybe we can create them in `support-test` similar to mocks or just add the `testImplementation` in app for mockito, similar to other common 3rd party dependencies.

Comment (rahulsainani) [reply android-components/components/support/test/build.gradle:40]: That's the general guidance. On further thought, `support-test` is intended to be a pass through module so let's move ahead with this. 👍 

Tangential
However, we should be mindful of `api`. Generally modules that need should directly `implement` 3rd party dependencies they need, as a result the modules remain fairly independent and can be compiled in parallel.

**Review by rahulsainani — APPROVED** [review]

File: android-components/components/support/test/build.gradle
Comment (MatthewTighe) [reply android-components/components/support/test/build.gradle:40]: Oh yeah, my intent was explicitly to use our existing support module as a passthrough. Apologies if that wasn't clear from the bug description.

On second thought though, should we remove the direct `implementation` usage throughout A-C as part of this patch?

Comment (rahulsainani) [reply android-components/components/support/test/build.gradle:40]: thanks for the clarification, sounds fair 👍

**Comment by MatthewTighe** [comment]
Realized it would be best to wait until Monday to land this, if anybody sees this and is wondering

**Pull request merged**
